overlord/ifacestate/handlers.go

```diff
@@ -1019,13 +1020,17 @@ func batchConnectTasks(st *state.State, snapsup *snapstate.SnapSetup, conns map[
 		opts.DelayedSetupProfiles = true
 		connectTs, err := connect(st, conn.PlugRef.Snap, conn.PlugRef.Name, conn.SlotRef.Snap, conn.SlotRef.Name, opts)
 		if err != nil {
-			return nil, fmt.Errorf("internal error: auto-connect of %q failed: %s", conn, err)
+			return nil, false, fmt.Errorf("internal error: auto-connect of %q failed: %s", conn, err)
+		}
+
+		if len(connectTs.Tasks()) > 1 {
```

Comment:
here we just check if the connect taskSet has connect hooks in it to determine that the snap has hooks, but then later, we use this variable to mean that the snap has specifically an install hook, so I guess my question is does `batchConnectTasks` return `hasHooks == true` for install hooks too? Because the implementation of `firstTaskAfterBootWhenPreseeding` suggests that we need to have an install hook so I'm confused if this is the right implementation or not. Basically I'm wondering if we are properly handling all of these cases:

1. snap has only an install hook
1. snap has only interface hooks
1. snap has both install hook and some interface hooks

because this implementation seems to only handle the third case

Reply:
I've renamed hasHooks variables to hasInterfaceHooks for clarity. We're only interested if any of the preseeded snaps has *interface hooks* (other hooks do not matter since they run on first boot, after preseeding).

If we detect interface hooks are present, then we re-order connect tasks and interface hooks: they run after mark-preseeded (i.e. on first boot), and before **install hooks** (which are normally run first after mark-preseeded).

Reply:
I see now that since we always have an install hook task, even if we don't actually have an install hook, that this logic makes sense and that the return value of this function is actually more precisely just about connect hooks (which is what the PR description/name says so :wink:)